retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/PipeBuffer.java

```diff
@@ -24,19 +27,21 @@
 import okio.Timeout;
 
 final class PipeBuffer {
-
+    @GuardedBy("reentrantLock")
     private final Buffer buffer = new Buffer();
     private final PipeSource source = new PipeSource();
     private boolean sinkClosed;
     private boolean sourceClosed;
+    private final ReentrantLock reentrantLock = new ReentrantLock();
     @Nullable
     private Throwable sinkClosedException;
 
     void write(byte[] source, int offset, int byteCount) {
         if (byteCount == 0) {
             return;
         }
-        synchronized (buffer) {
```

Comment:
It seems like `okio` enforces users to use `wait`/`notifyAll` so we can't simply replace `synchronized`. (causing test failure)

What do you think of reverting changes in this file?
https://github.com/square/okio/issues/1169

Reply:
got it. reverted changes in this file @jrhee17 
> It seems like `okio` enforces users to use `wait`/`notifyAll` so we can't simply replace `synchronized`. (causing test failure)
> 
> What do you think of reverting changes in this file? [square/okio#1169](https://github.com/square/okio/issues/1169)

Reply:
@ikhoon @jrhee17 I am facing the issue highlighted here [Task :examples:static-files:test FAILED](https://github.com/line/armeria/issues/1934) locally. any hint/idea?

Reply:
Do you have a stacktrace of the failure?

Reply:
<img width="1440" alt="Screenshot 2023-01-04 at 09 04 37" src="https://user-images.githubusercontent.com/36740791/210510681-93f61bd1-22cd-4310-9fc8-bd073b6d07cc.png">

Reply:
The test checks the user's home directory and tries to list the directory files.

https://github.com/line/armeria/blob/70296b99b4277eef75da93ec85eeb26beec8d39e/examples/static-files/src/main/java/example/armeria/server/files/Main.java#L42

Unfortunately, if an `index.html` file exists in the directory the file will be returned instead of a listing.

Can you check your home directory and try removing the `index.html` file?

Reply:
Thanks @jrhee17, after index.html deletion, test passed. 
Now 3 more new test failures as shown below even though server-list.properties actually exists.
<img width="1354" alt="Screenshot 2023-01-05 at 19 31 44" src="https://user-images.githubusercontent.com/36740791/
<img width="1413" alt="Screenshot 2023-
<img width="1423" alt="Screenshot 2023-01-05 at 19 30 06" src="https://user-images.githubusercontent.com/36740791/210857625-4289b82c-d384-403e-8630-45a214d17d45.png">
01-05 at 19 30 48" src="https://user-images.githubusercontent.com/36740791/210857609-884d9058-79c2-4fa3-9954-84913c4d3981.png">
210857583-0aae8824-5364-4291-b3a1-76bd1dafab4d.png">

Reply:
- `DefaultDnsCacheTest` seems like a flaky test so it could be ignored.
- For `PropertiesEndpointGroupTest`, it is not related to your change. Could you try to retest after running `gradle clean` task?

Reply:
I see, so what `shadedTest` does is basically
1. Shade dependencies into a jar
2. Unzip the jar into `"${project.buildDir}/classes/java/shaded-test"`
3. Run tests for classes in the above directory

Can you actually try to navigate to `"${project.buildDir}/classes/java/shaded-test"` and compare the file location with the one printed in the error logs?

My guess is the percent encoded url is causing problems, but I'd like to make sure first. If this is indeed the issue, I think it's better to handle this in a separate PR.

After handling the following comment, I'd like to get this PR merged.
https://github.com/line/armeria/pull/4610#discussion_r1060423181

Reply:
update) confirmed locally that the following patch fixes the issue.

```
-        final Path resourcePath = new File(resourceUrl.getFile()).toPath();
+        final Path resourcePath = new File(resourceUrl.toURI().getPath()).toPath();
```

Reply:
The patch works! PR here #4617

Reply:
@jrhee17 
> It seems like okio enforces users to use wait/notifyAll so we can't simply replace synchronized. (causing test failure)
Can I ask where okio enforces users to use wait/notifyAll?
I asking this because Okio change internal locks from `synchonized` to `ReentrantLock` since 3.3.0 like [this](https://github.com/square/okio/pull/1176/files#diff-1ee449bd262b427000deb65f08e41919132d29dd6631425a4e35b8e254aa9d1fL104).
The way they changed seems like just use `ReentrantLock`.